conan_provider.cmake

```diff
@@ -281,6 +287,23 @@ function(detect_compiler compiler compiler_version compiler_runtime compiler_run
         set(_compiler "clang")
         string(REPLACE "." ";" VERSION_LIST ${_compiler_version})
         list(GET VERSION_LIST 0 _compiler_version)
+
+        if(EMSCRIPTEN)
+            if(${conan_version} VERSION_GREATER_EQUAL "2.18.0")
```

Comment:
It is not necessary to add conditionals on the Conan version, otherwise this script will start being too complicated. Plus this cannot be easily tested.
We can assume that ``cmake-conan`` on the ``develop2`` branch, moves forward with the latest Conan 2.X.